### PR TITLE
Issue #143 Add codefix for S3005 (non static field with ThreadStaticAttribute)

### LINKED ISSUE #143: Add codefix for S3005 (non static field with ThreadStaticAttribute)

### PR DESCRIPTION
**Pull request by tvsonar**
Fixes #143

### CODE REVIEW

File: src/SonarLint/Rules/RedundantParenthesesCodeFixProvider.cs [hunk 1]
```diff
@@ -43,7 +43,7 @@ public sealed override FixAllProvider GetFixAllProvider()
         {
```
Comment (dbolkensteyn) [inline src/SonarLint/Rules/RedundantParenthesesCodeFixProvider.cs]: ahum this is completely unreleaded and should go into the redundant parentheses PR?

File: src/SonarLint/Rules/ThreadStaticNonStaticFieldCodeFixProvider.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            {
+                nodeToRemove = attributeList;
+            }
+
+            context.RegisterCodeFix(
+                CodeAction.Create(
+                    Title,
+                    c => RedundantParenthesesCodeFixProvider.RemoveUnusedCode(context.Document, nodeToRemove,
+                        SyntaxRemoveOptions.KeepNoTrivia, c),
+                    Title),
```
Comment (dbolkensteyn) [inline src/SonarLint/Rules/ThreadStaticNonStaticFieldCodeFixProvider.cs]: @tvsonar as discussed - let's drop this

File: src/SonarLint/Rules/ThreadStaticNonStaticFieldCodeFixProvider.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            var attributeList = nodeToRemove.Parent as AttributeListSyntax;
+            if (attributeList != null && attributeList.Attributes.Count == 1)
+            {
+                nodeToRemove = attributeList;
+            }
+
+            context.RegisterCodeFix(
+                CodeAction.Create(
+                    Title,
+                    c => RedundantParenthesesCodeFixProvider.RemoveUnusedCode(context.Document, nodeToRemove,
```
Comment (dbolkensteyn) [inline src/SonarLint/Rules/ThreadStaticNonStaticFieldCodeFixProvider.cs]: @tvsonar I don't think it's worth to reuse the code of `RedundantParenthesesCodeFixProvider` here - it's more misleading than helping. Both analyzers are completely unrelated

**Comment by dbolkensteyn** [comment]
LGTM

One small thing: On `[System.ThreadStatic, /*x*/ MTAThread /*y*/]` - we're losing the `/*x*/` comment - but that's minor

**Comment by tvsonar** [comment]
@dbolkensteyn The comment is a trailing trivia of the comma token. I don't think it is worth the trouble to keep it, because we would need to filter out the whitespace trivias, and only keep the  comment ones. It is going to be unnecessarily complex. (We already have to branch based on single/multiple attributes in the attribute list.

**Pull request merged**